src/QsCompiler/Core/SyntaxTreeTransformation.fs

```diff
@@ -170,58 +178,62 @@ and StatementTransformation<'T> internal (options, unsafe) =
+            this.Transformation <- new SyntaxTreeTransformation<'T>(sharedState)
             this.Transformation.Statements <- this
 
-    new (parentTransformation : QsSyntaxTreeTransformation<'T>) = 
+    new (parentTransformation : SyntaxTreeTransformation<'T>) = 
         new StatementTransformation<'T>(parentTransformation, TransformationOptions.Default)
     
-    new (sharedInternalState : 'T) = 
-        new StatementTransformation<'T>(sharedInternalState, TransformationOptions.Default)
+    new (sharedState : 'T) = 
+        new StatementTransformation<'T>(sharedState, TransformationOptions.Default)
 
 
-and NamespaceTransformation<'T> internal (options, unsafe : string) =
-    inherit NamespaceTransformationBase(options, unsafe)
-    let mutable _Transformation : QsSyntaxTreeTransformation<'T> option = None
+and NamespaceTransformation<'T> internal (options, _internal_ : string) =
+    inherit NamespaceTransformationBase(options, _internal_)
+    let mutable _Transformation : SyntaxTreeTransformation<'T> option = None
 
     member this.Transformation
         with get () = _Transformation.Value
         and private set value = 
             _Transformation <- Some value
             this.StatementTransformationHandle <- fun _ -> value.Statements :> StatementTransformationBase
 
-    new (parentTransformation : QsSyntaxTreeTransformation<'T>, options : TransformationOptions) as this = 
-        NamespaceTransformation<'T>(options, "unsafe") then
+    member this.SharedState = this.Transformation.SharedState
+
+    new (parentTransformation : SyntaxTreeTransformation<'T>, options : TransformationOptions) as this = 
+        NamespaceTransformation<'T>(options, "_internal_") then
             this.Transformation <- parentTransformation
 
-    new (sharedInternalState : 'T, options : TransformationOptions) as this =
-        NamespaceTransformation<'T>(options, "unsafe") then
-            this.Transformation <- new QsSyntaxTreeTransformation<'T>(sharedInternalState)
+    new (sharedState : 'T, options : TransformationOptions) as this =
+        NamespaceTransformation<'T>(options, "_internal_") then
+            this.Transformation <- new SyntaxTreeTransformation<'T>(sharedState)
             this.Transformation.Namespaces <- this
 
-    new (parentTransformation : QsSyntaxTreeTransformation<'T>) = 
+    new (parentTransformation : SyntaxTreeTransformation<'T>) = 
         new NamespaceTransformation<'T>(parentTransformation, TransformationOptions.Default)
     
-    new (sharedInternalState : 'T) = 
-        new NamespaceTransformation<'T>(sharedInternalState, TransformationOptions.Default)
+    new (sharedState : 'T) = 
+        new NamespaceTransformation<'T>(sharedState, TransformationOptions.Default)
 
 
 // setup for syntax tree transformations without internal state
 
-type QsSyntaxTreeTransformation private (unsafe : string) =
+type SyntaxTreeTransformation private (_internal_ : string) =
```

Comment:
It looks like the non-generic classes did not get the shortcut for SharedState.

Reply:
In other words, they are missing:
`member this.SharedState = this.Transformation.SharedState`

Reply:
The non-generic versions do not have a shared state.